app/org/sagebionetworks/bridge/services/ParticipantService.java

```
@@ -397,8 +397,6 @@ public void resendVerification(Study study, ChannelType type, String userId) {
         } else if (type == ChannelType.PHONE && participant.getPhone() != null) {
             AccountId accountId = AccountId.forPhone(study.getIdentifier(), participant.getPhone());
             accountWorkflowService.resendVerificationToken(type, accountId);
-        } else {
-            throw new UnsupportedOperationException("Channel type not implemented");
```

What's the problem with throwing this exception? If we're not passing in a channel, this is almost certainly a bug in the server. If this happens, it would be better to fast-fail the request with a 500 Internal Server Error than to do nothing and mask the error.

---

You get a 500 error back if the participant does not have an email or phone address, which just leaks information about the account. This isn't a pure test of the enumeration handling alone.

---

Why don't I throw a BadRequestException which says "this is not a valid request to resent a verification." This seems... vague enough not to indicate anything about the account.

---

Okay, I see it now. It's the compound clauses on the if statements. I think you need nested if statements.